File: algorithm/67. Add Binary/67.py
class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        i,j,c=len(a)-1,len(b)-1,0
        s=""
        while i>=0 or j>=0 or c==1:
            if i>=0:
                c += int(a[i])
            i-=1
            if j>=0:
                c +=int(b[j])
            j-=1
            s= str(c%2)+s
            c = c/2
        return s

File: algorithm/67. Add Binary/67.java
class Solution {
    public String addBinary(String a, String b) {
        int m = a.length() - 1, n = b.length() -1 , c = 0;
        StringBuilder sb = new StringBuilder();
        while(m>=0||n>=0||c>0)
        {
            c += (m>=0)?a.charAt(m--) - '0':0;
            c += (n>=0)?(b.charAt(n--) - '0'):0;
            sb.append(c%2);
            c/=2;
        }
        return sb.reverse().toString();
    }
}
